sound/soc/sof/pcm.c

```diff
@@ -370,14 +370,43 @@ static int sof_pcm_trigger(struct snd_pcm_substream *substream, int cmd)
 		return -EINVAL;
 	}
 
-	snd_sof_pcm_platform_trigger(sdev, substream, cmd);
+	/* DMA and IPC sequence is different for start and stop. Need to stop
+	 * DMA before STOP IPC sent.
```

Comment:
@xiulipan this comment is wrong based on the code you have here. For START trigger, you start DMA and then send START IPC. For STOP trigger, its the other way around.

Reply:
@ranj063 Thanks for point out the misleading comments. I almost got myself lost about who should go first.